src/data/data.h

```diff
@@ -13,8 +13,10 @@
 
 #include "xgboost/data.h"
 #include "simple_dmatrix.h"
-#include "sparse_page_dmatrix.h"
 #include "../common/group_data.h"
+#if DMLC_ENABLE_STD_THREAD
+#include "sparse_page_dmatrix.h"
+#endif  // DMLC_ENABLE_STD_THREAD
```

Comment:
Do we need this, if C++14 is required?

Reply:
It's defined in dmlc core so right now we have to align with it.

Reply:
I thought `DMLC_ENABLE_STD_THREAD` should only control whether `dmlc` uses `std::thread`? This line of code is not part of `dmlc`.

Reply:
Yup, but we have threaded iterator from dmlc-core.